app/graphql/resolvers/freaks_resolver.rb

```diff
@@ -2,11 +2,17 @@
 
 module Resolvers
   class FreaksResolver < Resolvers::Base
+    argument :filter, Types::FreakFilterType, required: false
+
     type Types::FreakType.connection_type, null: false
 
     # :reek:UtilityFunction
-    def resolve
-      Freak.all
+    def resolve(filter: nil)
+      query = Freak.all.group(:id).order(:id)
+      query = GetQuery.new.all_projects(filter, query)
+      query = GetQuery.new.any_project(filter, query)
+      query = GetQuery.new.all_technologies(filter, query)
+      GetQuery.new.any_technology(filter, query)
```

Comment:
Looks a bit better, but it can be even better.

There's lots of objects being created here - `GetQuery.new` x3 + each of them does `QueryFilter.new` as well. 

There's also lots of repetition - try to find a solution in which you'd pass the filter & query and it would give you the correct query, leaving you with 2 lines of code here.

See if the Builder pattern could help https://refactoring.guru/design-patterns/builder.